bindings/nodejs/src/functions/spi.cc

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <node.h>
+#include <nan.h>
+#include <sol-spi.h>
+
+#include "../common.h"
+#include "../data.h"
+#include "../hijack.h"
+#include "../structures/js-handle.h"
+#include "../structures/sol-js-spi.h"
+
+using namespace v8;
+
+class SolSpi : public JSHandle<SolSpi> {
+public:
+    static const char *jsClassName() { return "SolSpi"; }
+};
+
+NAN_METHOD(bind_sol_spi_open) {
+    VALIDATE_ARGUMENT_COUNT(info, 2);
+    VALIDATE_ARGUMENT_TYPE(info, 0, IsUint32);
+    VALIDATE_ARGUMENT_TYPE(info, 1, IsObject);
+
+    sol_spi_config config;
+    sol_spi *spi = NULL;
+
+    if (!c_sol_spi_config(info[1]->ToObject(), &config)) {
+        return;
+    }
+
+    spi = sol_spi_open(info[0]->Uint32Value(), &config);
+    if (spi) {
+        info.GetReturnValue().Set(SolSpi::New(spi));
+    }
+}
+
+NAN_METHOD(bind_sol_spi_close) {
+    VALIDATE_ARGUMENT_COUNT(info, 1);
+    VALIDATE_ARGUMENT_TYPE_OR_NULL(info, 0, IsObject);
+    Local<Object> jsSpi = Nan::To<Object>(info[0]).ToLocalChecked();
+    sol_spi *spi= (sol_spi *)SolSpi::Resolve(jsSpi);
+    if (!spi)
+        return;
+
+    sol_spi_close(spi);
+    Nan::SetInternalFieldPointer(jsSpi, 0, 0);
+}
+
+static void sol_spi_transfer_cb(void *cb_data, struct sol_spi *spi,
+    const uint8_t *tx, uint8_t *rx, ssize_t status) {
+    Nan::HandleScope scope;
+    Nan::Callback *callback = (Nan::Callback *)cb_data;
+    Local<Value> txBuffer;
+    Local<Value> rxBuffer;
+
+    if (status >= 0) {
+        txBuffer = Nan::NewBuffer((char *)tx, status).ToLocalChecked();
+        rxBuffer = Nan::NewBuffer((char *)rx, status).ToLocalChecked();
+    } else {
+        txBuffer = Nan::Null();
+        rxBuffer = Nan::Null();
```

Comment:
@cabelitos Good point. Instead of checking the _status_, it make sense to check the _tx_ and _rx_ and pass the ownership to NewBuffer if not null. I will make this change in a new PR.